firebase-appdistribution/test-app/src/main/java/com/googletest/firebase/appdistribution/testapp/ScreenshotDetectionFeedbackTrigger.kt

```diff
@@ -0,0 +1,226 @@
+          if (!isEnabled) {
+            warn("Trigger disabled after permission check. Abandoning screenshot detection.")
+          } else if (isGranted) {
+            maybeStartFeedbackForScreenshot(activity, currentUri!!)
+          } else {
+            info("Permission not granted")
+            // TODO: Ideally we would show a message indicating the impact of not enabling
+            // the permission, but there's no way to know if they've permanently denied
+            // the permission, and we don't want to show them a message after every
+            // screenshot.
+          }
+        }
+    } else {
+      warn("Not listening for screenshots because this activity can't register for permission request results: $activity")
+    }
+  }
+
+  override fun onActivityResumed(activity: Activity) {
+    currentActivity = activity
+    if (isEnabled) {
+      listenForScreenshots()
+    }
+  }
+
+  override fun onActivityPaused(activity: Activity) {
+    if (isEnabled) {
+      stopListeningForScreenshots()
+    }
+    currentActivity = null
+  }
+
+  private fun requestReadPermission(activity: Activity, uri: Uri) {
+    if (activity.shouldShowRequestPermissionRationale(permissionToRequest)) {
+      info("Showing customer rationale for requesting permission.")
+      AlertDialog.Builder(activity)
+        .setMessage(
+          "Taking a screenshot of the app can initiate feedback to the developer. To enable this feature, allow the app access to device storage."
+        )
+        .setPositiveButton("OK") { _, _ ->
+          info("Launching request for permission.")
+          currentUri = uri
+          requestPermissionLauncher!!.launch(permissionToRequest)
+        }
+        .show()
+    } else {
+      info("Launching request for permission without rationale.")
+      currentUri = uri
+      requestPermissionLauncher!!.launch(permissionToRequest)
+    }
+  }
+
+  private fun maybeStartFeedbackForScreenshot(activity: Activity, uri: Uri) {
+    var cursor: Cursor? = null
+    try {
+      cursor = activity.contentResolver.query(uri, contentProjection, null, null, null)
+      if (cursor != null && cursor.moveToFirst()) {
+        val pendingColumn = cursor.getColumnIndexOrThrow(MediaStore.MediaColumns.IS_PENDING)
+        if (shouldCheckIfPending && cursor.getInt(pendingColumn) == 1) {
+          info("Ignoring pending image: $uri")
+          return
+        }
+        seenImages.add(uri)
+        val path = cursor.getString(cursor.getColumnIndexOrThrow(MediaStore.Images.Media.DATA))
+        info("Path: $path")
+        if (path.lowercase(Locale.getDefault()).contains("screenshot")) {
+          FirebaseAppDistribution.getInstance().startFeedback(infoTextResourceId, uri)
+        }
+      }
+    } catch (e: Exception) {
+      error("Could not determine if media change was due to taking a screenshot", e)
+    } finally {
+      cursor?.close()
```

Comment:
it looks like `Cursor` implements `Closeable`, so you should be able to use try-with-resources. 
Oops, this is not Java - of course I mean `use` ([doc](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/use.html))

Reply:
Oh nice!